test/sass/mixins--spacing.spec.scss

```diff
@@ -434,3 +434,602 @@
+        @include margin(0, "inline");
+      }
+
+      @include expect {
+        margin-left: 0;
+        margin-right: 0;
+        margin-inline: 0;
+      }
+
+    }
+  }
+
+  @include it("generates correct fallbacks with a single non-zero value for dimension 'inline'") {
+    @include assert() {
+
+      @include output {
+        @include margin(16, "inline");
+      }
+
+      @include expect {
+        margin-left: 16px;
+        margin-left: 1rem;
+        margin-right: 16px;
+        margin-right: 1rem;
+        margin-inline: 1rem;
+      }
+
+    }
+  }
+
+  @include it("generates correct fallbacks with a pair of values for dimension 'inline'") {
+    @include assert() {
+
+      @include output {
+        @include margin(16 32, "inline");
+      }
+
+      @include expect {
+        html[dir="ltr"] &:not([dir]),
+        &[dir="ltr"] {
+          margin-left: 16px;
+          margin-left: 1rem;
+          margin-right: 32px;
+          margin-right: 2rem;
+        }
+        html[dir="rtl"] &:not([dir]),
+        &[dir="rtl"] {
+          margin-right: 16px;
+          margin-right: 1rem;
+          margin-left: 32px;
+          margin-left: 2rem;
+        }
+
+        html[dir] & {
+          margin-inline: 1rem 2rem;
+        }
+      }
+
+    }
+
+    @include assert() {
+
+      @include output {
+        @include margin(16 16, "inline");
+      }
+
+      @include expect {
+        margin-left: 16px;
+        margin-left: 1rem;
+        margin-right: 16px;
+        margin-right: 1rem;
+        margin-inline: 1rem;
+      }
+
+    }
+
+    @include assert() {
+
+      @include output {
+        @include margin(0 32, "inline");
+      }
+
+      @include expect {
+        html[dir="ltr"] &:not([dir]),
+        &[dir="ltr"] {
+          margin-left: 0;
+          margin-right: 32px;
+          margin-right: 2rem;
+        }
+        html[dir="rtl"] &:not([dir]),
+        &[dir="rtl"] {
+          margin-right: 0;
+          margin-left: 32px;
+          margin-left: 2rem;
+        }
+
+        html[dir] & {
+          margin-inline: 0 2rem;
+        }
+      }
+
+    }
+
+    @include assert() {
+
+      @include output {
+        @include margin(16 0, "inline");
+      }
+
+      @include expect {
+        html[dir="ltr"] &:not([dir]),
+        &[dir="ltr"] {
+          margin-left: 16px;
+          margin-left: 1rem;
+          margin-right: 0;
+        }
+        html[dir="rtl"] &:not([dir]),
+        &[dir="rtl"] {
+          margin-right: 16px;
+          margin-right: 1rem;
+          margin-left: 0;
+        }
+
+        html[dir] & {
+          margin-inline: 1rem 0;
+        }
+      }
+
+    }
+
+    @include assert() {
+
+      @include output {
+        @include margin(0 0, "inline");
+      }
+
+      @include expect {
+        margin-left: 0;
+        margin-right: 0;
+        margin-inline: 0;
+        }
+
+    }
+  }
+
+  @include it("Ignores more than two values in the sizes list for the dimension 'inline'") {
```

Comment:
Makes me think it might be better to have more explicit mixins rather than having arguments that change what they do (eg `@include margin-inline(16, 32)`).

Reply:
I see the point, but not sure it's the right thing to do here: I'd rather have just one way of setting margins, at least to start off with. Using @stephenwf's idea of throwing errors when there are an incorrect number of values passed will supply rapid developer feedback, preventing the silent fail which this code currently allows.